### DISCUSSION TITLE
How does DeepExplainer works in SHAP?

### DISCUSSION THREAD

**Discussion by khanfarabi** [shap/shap#1938] [Q&A]
Hi, I am using SHAP to generate explanation of the Deep Network prediction. I need to understand how DeepExplainer works? Is it possible explain the way DeepExplainer works to generate the explanations? 

Thanks.

**Comment by Lakshya-CB** [comment]
Hello, Khanfarabi.
try reading this https://medium.com/@gabrieltseng/interpreting-complex-models-with-shap-values-1c187db6ec83
https://proceedings.neurips.cc/paper/2017/file/8a20a8621978632d76c43dfd28b67767-Paper.pdf

**Comment by khanfarabi** [comment]
Hi,

Thanks for your reply. I was trying to check sentiment analysis using SHAP from your following notebook:
https://nbviewer.jupyter.org/github/slundberg/shap/blob/master/notebooks/text_examples/sentiment_analysis/Keras%20LSTM%20for%20IMDB%20Sentiment%20Classification.ipynb

However, I am getting error at:
shap_values = explainer.shap_values(x_test[:10])

Error is: NoneType' object cannot be interpreted as an integer


Is it possible to help in this?

Yours Truly,
Khan
________________________________
